### ISSUE TITLE
[opentelemetry-php-contrib] Automatically associate monolog record to the current trace from context

### ISSUE THREAD

**Issue by DominicDetta** [open-telemetry/opentelemetry-php#1441]
I'm using the `open-telemetry/opentelemetry-logger-monolog` package but log records generated are not related to request trace.
Following is a screenshot from the [Signoz](https://github.com/SigNoz/signoz) monitoring solution:
<img width="1001" alt="Screenshot 2024-11-27 alle 16 24 20" src="https://github.com/user-attachments/assets/c93a5c01-17b9-42c8-ba06-fed4101d1e76">

Is this a bug or a normal behaviour?

Without the trace_id the APM is not able to navigate automatically to the associated request trace.

From the example of this [page](https://github.com/open-telemetry/opentelemetry-php/blob/main/examples/logs/getting_started.php) I think it should be possible to inject the trace and span id automatically

**Comment by brettmc** [comment]
Hi @DominicDetta there is some configuration required for that package, and it's not clear whether you have done it or not. You need to create a handler and configure monolog to use it, and the otel logger provider can be either manually or automatically created. 

Have you tried something like in the [examples](https://github.com/open-telemetry/opentelemetry-php-contrib/tree/main/src/Logs/Monolog/example)?

**Comment by DominicDetta** [comment]
Hi, this is the code added to the monolog class:

**Comment by DominicDetta** [comment]
Maybe the exception is triggered before the instrumentation is able to create the trace id?

**Comment by brettmc** [comment]
> Maybe the exception is triggered before the instrumentation is able to create the trace id?

That's a good point - it could be that the error is logged either before or after the trace. Do you use auto-instrumentation or manual in your code?

**Comment by brettmc** [comment]
@DominicDetta is this still an issue for you?

**Comment by DominicDetta** [comment]
Yes, the problem appears but at the moment it's not crucial to the application.

Maybe it's related on how the framework works.
 I'm using [Mezzio](https://docs.mezzio.dev/)

**Comment by DominicDetta** [comment]
I will close it
